File: node/currying.js
function log(level, message) {
    console.log(`[${level}]: ${message}`)
}

function logForLevel(level) {
    return function (message) {
        log(level, message)
    }
}

const logDebug = logForLevel('DEBUG')
logDebug('Hello, World!')  // [DEBUG] Hello, World!

File: golang/src/currying.go
package main

import "fmt"

func log(level string, message string) {
	fmt.Printf("[%s]: %s\n", level, message)
}

func logForLevel(level string) func(string) {
	return func(message string) {
		log(level, message)
	}
}

func main() {
	logDebug := logForLevel("DEBUG")
	logDebug("Hello, World!") // [DEBUG] Hello, World!
}
